src/lib.rs

```diff
@@ -169,6 +106,30 @@ impl Default for MaybeInitializedCaptureIndex {
 
 const ALL_NODES_QUERY: &str = "(_) @node";
 
+#[derive(Default)]
+struct CachedQueries(Mutex<HashMap<SupportedLanguageName, Option<Arc<Query>>>>);
+
+impl CachedQueries {
+    fn get_and_cache_query_for_language(
+        &self,
+        query_source: &str,
+        language: &dyn SupportedLanguage,
+    ) -> Option<Arc<Query>> {
+        self.0
+            .lock()
+            .unwrap()
+            .entry(language.name())
+            .or_insert_with(|| maybe_get_query(query_source, language.language()).map(Arc::new))
+            .clone()
+    }
```

Comment:
This seems like a pretty reasonable use of interior mutability. Since the thing that requires the lock is predictable (it's never going to block for seconds or something), it seems fine to completely hide that from the consumer to me.

Reply:
Ya (per the other comment re updating to use a per-language `OnceLock`) this now seems even more (per I believe a previous comment that "this really shouldn't be that non-optimal because you shouldn't ever have to wait much longer than how long it takes to perform the actual query which you could have to do regardless") like you won't have to wait longer than that since I think the behavior of `OnceLock` is that once it's been initialized then calls to `.get_or_init()` won't block?

Reply:
Yep, this seems perfect.